lib/chef/resource/windows_ad_join.rb

```diff
@@ -86,6 +86,17 @@ def on_domain?
           raise "Failed to check if the system is joined to the domain #{new_resource.domain_name}: #{node_domain.stderr}}" if node_domain.error?
           node_domain.stdout.downcase.strip == new_resource.domain_name.downcase
         end
+
+        def clarify_reboot(reboot_action)
```

Comment:
Maybe a comment to why we're actually doing this also. Makes sense in this context, but might not be entirely clear in a few years